matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/crypto/model/OlmInboundGroupSessionWrapperOld.kt

```diff
@@ -119,18 +115,14 @@ internal class OlmInboundGroupSessionWrapper2 : Serializable {
                 return null
             }
 
-            val safeOlmInboundGroupSession = olmInboundGroupSession ?: return null
-
-            val wantedIndex = index ?: safeOlmInboundGroupSession.firstKnownIndex
-
             MegolmSessionData(
                     senderClaimedEd25519Key = keysClaimed?.get("ed25519"),
-                    forwardingCurve25519KeyChain = forwardingCurve25519KeyChain?.toList().orEmpty(),
+                    forwardingCurve25519KeyChain = ArrayList(forwardingCurve25519KeyChain!!),
                     senderKey = senderKey,
                     senderClaimedKeys = keysClaimed,
                     roomId = roomId,
-                    sessionId = safeOlmInboundGroupSession.sessionIdentifier(),
-                    sessionKey = safeOlmInboundGroupSession.export(wantedIndex),
+                    sessionId = olmInboundGroupSession!!.sessionIdentifier(),
```

Comment:
:100: it will be interesting to see if we were swallowing errors by returning null earlier without any logging